drivers/timer/nrf_rtc_timer.c

```diff
@@ -371,25 +362,19 @@ void _enable_sys_clock(void)
 u64_t _get_elapsed_clock_time(void)
 {
 	u64_t elapsed;
-	u32_t rtc_now, rtc_elapsed, rtc_prev, sys_elapsed;
+	u32_t rtc_elapsed, rtc_past_copy;
 
-	rtc_now = RTC_COUNTER;
-
-	/* Discard value of RTC_COUNTER read at LFCLK transition */
-	do {
-		/* Calculate number of rtc cycles elapsed since RTC programing*/
-		rtc_elapsed = (rtc_now - rtc_past) & RTC_MASK;
-		elapsed = _sys_clock_tick_count;
-		rtc_prev = rtc_now;
-		rtc_now = RTC_COUNTER;
-	} while (rtc_now != rtc_prev);
+	/* Read _sys_clock_tick_count and rtc_past before RTC_COUNTER */
+	elapsed = _sys_clock_tick_count;
+	rtc_past_copy = rtc_past;
+	compiler_barrier();
```

Comment:
Why do we need the compiler_barrier here? should we add an inline comment that justifies it?

Reply:
According to @cvinayak, the rtc_past must be read before RTC_COUNTER. Barrier ensures the correct order.

Reply:
I don't see the problem of placing the same comment in 2 locations in a file. It is really educative in this case.

Reply:
Ok

Reply:
Fixed.